api/features/workflows/core/models.py

```diff
@@ -179,6 +180,30 @@ def _publish_change_sets(self, published_by: "FFAdminUser") -> None:
         for change_set in self.change_sets.all():
             change_set.publish(user=published_by)
 
+    def _publish_segments(self) -> None:
+        segments = list(self.segments.all())
+        if not segments:
+            return
+
+        for segment in segments:
+            target_segment = segment.version_of
+            assert target_segment != segment
+
+            # Deep clone the segment to establish historical version.
+            target_segment.deep_clone()
```

Comment:
Couldn't we optimise this by sending in an argument, or via a separate method, which doesn't clone the target_segments rules? Since we're just deleting them again below, it seems unnecessary to do this.

Reply:
This is for establishing a versioned history for the soon-to-be-changing segment. Without a deep clone we won't have a copy of a previous version of the segment with the associated rules and conditions. We can always skip doing the deep clone but then our version history would have gaps in it if change requests are used to change segments.

Reply:
I don't think that's true is it? I get that we want to keep the history, but we don't need to actually duplicate the rules and conditions, we just need to archive the ones from the current (outgoing) segment. We don't need to create them for the new segment because we're not going to use them - we just delete them below.

Reply:
As discussed, this should be thought of like regular updates to a segment where we create the clone to create the version, then modifying the new 'draft' version with the data from the change request. 

Let's update the comments for this method to try and make this a little clearer.

Reply:
Ok the comment has been updated.